### PR TITLE
Changes to make Nop more accessible for disabled customers.

### PR DESCRIPTION
**Pull request by w-tbury**
This is an initial change to start making Nop Commerce more accessible to disabled customers.  These changes will help with screen readers accessing the Nop Commerce web site.  
These changes are based on the Chrome accessibility auditing tool results ran on the base Nop Commerce web site.
Changes include:
- Adding alt text to images.
- Adding aria-labels to input forms and links for those items that do not already have a descriptive labels or text.
- Setting the default lang attribute on the html tag depending on current language selected.

As testing continues, there will be more areas that may need to be modified to make the pages compliant with accessibility laws in various countries.  Any changes made were made to have the least impact on existing code.

### CODE REVIEW

File: src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs [hunk 1]
```diff
@@ -48,18 +48,23 @@ public IActionResult Configure()
                 Picture1Id = nivoSliderSettings.Picture1Id,
                 Text1 = nivoSliderSettings.Text1,
                 Link1 = nivoSliderSettings.Link1,
+                AltText1 = nivoSliderSettings.AltText1,
```
Comment (mariannk) [inline src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: Formatting issue: you need to add an extra line between blocks as it was done before.

File: src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs [hunk 2]
```diff
@@ -68,18 +73,23 @@ public IActionResult Configure()
                 model.Picture1Id_OverrideForStore = _settingService.SettingExists(nivoSliderSettings, x => x.Picture1Id, storeScope);
                 model.Text1_OverrideForStore = _settingService.SettingExists(nivoSliderSettings, x => x.Text1, storeScope);
                 model.Link1_OverrideForStore = _settingService.SettingExists(nivoSliderSettings, x => x.Link1, storeScope);
+                model.AltText1_OverrideForStore = _settingService.SettingExists(nivoSliderSettings, x => x.AltText1, storeScope);
```
Comment (mariannk) [inline src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: The same formatting issue as previous

File: src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs [hunk 3]
```diff
@@ -108,38 +118,48 @@ public IActionResult Configure(ConfigurationModel model)
             nivoSliderSettings.Picture1Id = model.Picture1Id;
             nivoSliderSettings.Text1 = model.Text1;
             nivoSliderSettings.Link1 = model.Link1;
+            nivoSliderSettings.AltText1 = model.AltText1;
```
Comment (mariannk) [inline src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: The same formatting issue as previous

**Comment by w-tbury** [comment]
Reviewed the merge conflicts and resolved.

**Comment by mariannk** [comment]
Hi! I noticed you have fixed the merge conflict i mentioned before. Thank you.
Will there be any updates regarding other issues I mentioned in comments?

**Comment by w-tbury** [comment]
I've looked through the pull request and I am not seeing any line item comments on the files.  Could you please add them or just add the comments here so that I can make those requested changes.  Thanks.

**Comment by mariannk** [comment]
Oh I'm sorry I should have clicked a common "Submit" button after all comments are done. I hope you see them now!

File: src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs [hunk 4]
```diff
@@ -108,38 +118,48 @@ public IActionResult Configure(ConfigurationModel model)
...
             nivoSliderSettings.Link5 = model.Link5;
+            nivoSliderSettings.AltText5 = model.AltText5;
 
             /* We do not clear cache after each setting update.
              * This behavior can increase performance because cached settings will not be cleared 
              * and loaded from database after each update */
             _settingService.SaveSettingOverridablePerStore(nivoSliderSettings, x => x.Picture1Id, model.Picture1Id_OverrideForStore, storeScope, false);
             _settingService.SaveSettingOverridablePerStore(nivoSliderSettings, x => x.Text1, model.Text1_OverrideForStore, storeScope, false);
             _settingService.SaveSettingOverridablePerStore(nivoSliderSettings, x => x.Link1, model.Link1_OverrideForStore, storeScope, false);
+            _settingService.SaveSettingOverridablePerStore(nivoSliderSettings, x => x.AltText1, model.AltText1_OverrideForStore, storeScope, false);
```
[the remark this answers is not in the document]
Comment (w-tbury) [reply src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: Not sure what previous issue you mention here.

**Comment by w-tbury** [comment]
Made all changes requested except for the last comment.  Not sure of what formatting issue you are referring to.

File: src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mariannk) [reply src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: I'm sorry for these confusing comments! Forget about it

[the remark this answers is not in the document]
Comment (mariannk) [reply src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: Extra lines between block are only in WidgetsNivoSliderViewComponent and not in Controller

[the remark this answers is not in the document]
Comment (w-tbury) [reply src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: No problem, I will go and make the change to WidgetsNivoSliderViewComponent and submit that.  Thanks.

[the remark this answers is not in the document]
Comment (w-tbury) [reply src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: I followed the same spacing conventions that already existed within the files changed.  Did you want me to make a change to that?  If not, it looks like I have addressed all issues.

[the remark this answers is not in the document]
Comment (w-tbury) [reply src/Plugins/Nop.Plugin.Widgets.NivoSlider/Controllers/WidgetsNivoSliderController.cs]: Just following up to see where the pull request is at.  Please advise.

**Comment by mariannk** [comment]
Hi!
I merged your pull-request. I reviewed the code one more time and made required changes by myself.
Please see the commit: https://github.com/nopSolutions/nopCommerce/commit/a7ab9fa5462ecea1e190c3d8a630e7d3b47584ec

You can see that I removed such labels as
aria-label="@T("ShoppingCart.AddToCart.CustomerEnteredPrice.Label")"
aria-label="@T("ShoppingCart.AddToCart.Quantity.Label")"
aria-label="@T("ShoppingCart.Quantity.Label")"

since related inputs already had labels.

**Pull request merged**